File: Desktop/EY/aitax-poc/apps/backend/app/models/news.py
from datetime import datetime
from enum import Enum
from typing import List, Optional

from sqlmodel import Field, SQLModel, Relationship


class TaxCategory(str, Enum):
    """Tax categories for news items"""
    VAT = "VAT"
    CIT = "CIT"
    PIT = "PIT"
    TRANSFER_PRICING = "Transfer Pricing"
    TAX_PROCEDURE = "Tax Procedure"
    INTERNATIONAL_TAX = "International Tax"
    OTHER = "Other"


class NewsBase(SQLModel):
    """Base news model"""
    title: str
    content: str
    summary: str
    category: TaxCategory
    source_url: Optional[str] = None
    published_date: datetime


class News(NewsBase, table=True):
    """News model for database"""
    id: Optional[int] = Field(default=None, primary_key=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    # Relationships
    notes: List["Note"] = Relationship(back_populates="news")


class NewsCreate(NewsBase):
    """News creation model"""
    pass


class NewsRead(NewsBase):
    """News read model"""
    id: int
    created_at: datetime


class NewsUpdate(SQLModel):
    """News update model"""
    title: Optional[str] = None
    content: Optional[str] = None
    summary: Optional[str] = None
    category: Optional[TaxCategory] = None
    source_url: Optional[str] = None
    published_date: Optional[datetime] = None

File: Desktop/EY/aitax-poc/apps/frontend/src/types/news.ts
export enum TaxCategory {
  VAT = "VAT",
  CIT = "CIT",
  PIT = "PIT",
  TRANSFER_PRICING = "Transfer Pricing",
  TAX_PROCEDURE = "Tax Procedure",
  INTERNATIONAL_TAX = "International Tax",
  OTHER = "Other"
}

export interface NewsBase {
  title: string;
  content: string;
  summary: string;
  category: TaxCategory;
  source_url?: string;
  published_date: string;
}

export interface News extends NewsBase {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface PersonalizedNews {
  news_id: number;
  original_summary: string;
  personalized_summary: string;
}
